scripts/azureml-assets/azureml/assets/release/asset_publish.py

```diff
@@ -231,49 +232,58 @@ def _str2bool(v: str) -> bool:
     logger.print(f"create list: {publish_list}")
 
     failure_list = []
-    for asset in util.find_assets(input_dirs=assets_dir):
-        asset_names = publish_list.get(asset.type.value, [])
-        if not ("*" in asset_names or asset.name in asset_names):
-            logger.print(
-                f"Skipping registering asset {asset.name} because it is not in the publish list")
+    all_assets = util.find_assets(input_dirs=assets_dir)
+    assets_by_type = {}
+    for asset in all_assets:
+        assets_by_type[asset.type.value] = assets_by_type.get(asset.type.value, []).append(asset)
+
+    for publish_asset_type in PUBLISH_ORDER:
+        logger.print(f"now publishing {publish_asset_type.value}s.")
+        if publish_asset_type.value not in publish_list:
             continue
```

Comment:
Consider removing these since the `for` below will do the same thing.

Reply:
What if a type in `PUBLISH_ORDER` is not in `publish_list`?
Will the for-loop throw errors if iterate an empty list?

Reply:
Then it will hit this part:
```
logger.print(
    f"Skipping registering asset {asset.name} because it is not in the publish list")
    continue
```

Reply:
Up to you... You can leave it in as a shortcut.